### PR TITLE
Add MACE potential

### PR DESCRIPTION
**Pull request by vitartas**
MACE potential was implemented based on run_train.py script (https://github.com/ACEsuit/mace/blob/main/scripts/run_train.py). install_mace.py script was added as well.

### CODE REVIEW

File: mlptrain/potentials/mace/mace.py [hunk 1]
```diff
@@ -0,0 +1,729 @@
...
+        return config_type_weights
+
+    @property
+    def z_table(self) -> 'mace.tools.AtomicNumberTable':
+        """Table containing atomic numbers of the system"""
+
+        if self._z_table is None:
+            self._z_table = tools.get_atomic_number_table_from_zs(
+                z
+                for configs in (self.train_configs, self.valid_configs)
```
Comment (tristan-j-wood) [inline mlptrain/potentials/mace/mace.py:304]: Could also consider explicit nested for loops for readability

File: mlptrain/potentials/mace/mace.py [hunk 2]
```diff
@@ -0,0 +1,729 @@
...
+    @property
+    def atomic_energies_array(self) -> np.ndarray:
+        """List of atomic energies of the system"""
+        return np.array([self.atomic_energies[symbol]
+                         for symbol in self.z_table_symbol])
+
+    @property
+    def valid_fraction(self) -> float:
+        """Fraction of the whole dataset to be used as validation set"""
+        _min_dataset = -(1 // -Config.mace_params['valid_fraction'])
```
Comment (tristan-j-wood) [inline mlptrain/potentials/mace/mace.py:324]: Add space here

File: mlptrain/potentials/mace/mace.py [hunk 3]
```diff
@@ -0,0 +1,729 @@
...
+                    for config in self.valid_configs],
+                batch_size=self.valid_batch_size,
+                shuffle=False,
+                drop_last=False)
+
+        return self._valid_loader
+
+    @property
+    def avg_num_neighbors(self) -> float:
+        """Average number of neighbours in the training set"""
```
Comment (tristan-j-wood) [inline mlptrain/potentials/mace/mace.py:457]: Keep neighbours spelling consistent

File: mlptrain/potentials/mace/mace.py [hunk 4]
```diff
@@ -0,0 +1,729 @@
...
+                },
+            ],
+            lr=self.args.lr,
+            amsgrad=Config.mace_params['amsgrad'])
+
+        return param_options
+
+    @property
+    def optimizer(self) -> 'torch.optim.Optimizer':
+        """Optimiser to use in training"""
```
Comment (tristan-j-wood) [inline mlptrain/potentials/mace/mace.py:608]: Optimizer

File: mlptrain/potentials/mace/mace.py [hunk 5]
```diff
@@ -0,0 +1,729 @@
...
+    def swa(self) -> Optional['mace.tools.SWAContainer']:
+        """Object for stochastic weight averaging during training"""
+
+        if self._swa is None:
+
+            if Config.mace_params['swa']:
+
+                if Config.mace_params['start_swa'] is None:
+                    # if not set start swa at 75% of training
+                    start_swa = self.max_num_epochs // 4 * 3
```
Comment (tristan-j-wood) [inline mlptrain/potentials/mace/mace.py:683]: Add bracket

File: mlptrain/potentials/mace/mace.py [hunk 6]
```diff
@@ -0,0 +1,729 @@
...
+        delta_time = time.perf_counter() - start_time
+
+        logger.info(f'MACE training ran in {delta_time / 60:.1f} m')
+
+        self._load_latest_epoch()
+        self._print_error_table()
+        self._save_model()
+        self._reset_train_objs()
+
+        os.remove(f'{self.name}_data.xyz')
```
[the remark this answers is not in the document]
Comment (vitartas) [reply mlptrain/potentials/mace/mace.py:110]: Probably can keep this line as it's done for other MLPs as well, .xyz data is saved automatically only when using al_train():
`    def al_train(self,
                 method_name: str,
                 **kwargs
                 ) -> None:
        """
        Train this MLP using active learning (AL) using a defined reference
        method

        -----------------------------------------------------------------------
        Arguments:
            method_name:

            **kwargs:  Keyword arguments passed to mlt.training.active.train()
        """
        al_train(self, method_name=method_name, **kwargs)
        self._save_training_data_as_npz_and_xyz()

        return None`

**Review by tristan-j-wood — APPROVED** [review]

**Pull request merged**